assets/src/edit-story/components/panels/textStyle/color.js

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useCallback, useEffect, useState } from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { Color, Label, Row } from '../../form';
+import getCommonValue from '../utils/getCommonValue';
+
+function ColorControls({ selectedElements, onSetProperties }) {
+  const color = getCommonValue(selectedElements, 'color');
+  const backgroundColor = getCommonValue(selectedElements, 'backgroundColor');
+
+  const [state, setState] = useState({
+    backgroundColor,
+    color,
+  });
+  useEffect(() => {
+    setState({
+      backgroundColor,
+      color,
+    });
+  }, [color, backgroundColor]);
+
+  const updateProperties = useCallback(() => {
+    onSetProperties(state);
+  }, [onSetProperties, state]);
+
+  useEffect(() => {
+    updateProperties();
+  }, [state.backgroundColor, state.color, updateProperties]);
```

Comment:
Isn't this the thing that caused infinite loop last time?

/cc @wassgha

Reply:
This shouldn't case an infinite loop, see the [comment here](https://github.com/google/web-stories-wp/pull/525#issuecomment-597592191) and [also the PR](https://github.com/google/web-stories-wp/pull/534) for restoring the logic, PTAL.